src/Umbraco.SampleSite/Controllers/LessonsController.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+using System.Web.Http;
+using Newtonsoft.Json;
+using Umbraco.Core;
+using Umbraco.Core.Cache;
+using Umbraco.Core.Configuration;
+using Umbraco.Core.Logging;
+using Umbraco.Web.Editors;
+using Umbraco.Web.Mvc;
+using Umbraco.Web.WebApi.Filters;
+
+namespace Umbraco.SampleSite.Controllers
+{
+    [PluginController("Starterkit")]
+    public class LessonsController : UmbracoAuthorizedJsonController
+    {
+        /// <summary>
+        /// Fetches available lessons for a given section from our.umbaco.org
+        /// </summary>
+        /// <param name="path">Name of the documentation section to fetch from, ex: "getting-started", "Tutorials/Starter-kit/Lessons" </param>
+        /// <param name="baseUrl">The url to retrieve from, by default https://our.umbraco.org</param>
+        /// <returns></returns>
+        [ValidateAngularAntiForgeryToken]
+        public async Task<IEnumerable<Lesson>> GetLessons(string path, string baseUrl = "https://our.umbraco.org")
```

Comment:
So basically since this is public, any person on the internet could tell this app to go fetch content from a totally different baseUrl and provide it some horrible content, perhaps even malicious, in it's simplest form you could have content full of profanity. Then this gets returned to this site and cached and then someone logs in to the back office and now seas this fake crappy, potentially malicious content.

.... Oh wait. Phew, i just noticed the `ValidateAngularAntiForgeryToken` , ok ignore me... but i'll leave this note here anyways since it's important to think about these things :)